💥 Only include dev deps if option set

grdryn opened the pull request:
This introduces an '--include-dev-dependencies' option to the console,
report, and save commands. The default behaviour changes here also to
include only production dependencies by default.

Fixes #241.

grdryn commented on lib/commands/console.cmd.js:
```diff
@@ -35,6 +35,11 @@ module.exports = () => {
       describe: 'Will list all production licenses for all modules.',
       default: false
+    },
+    'production-only': {
```
If you think this should be the default behaviour, but still configurable by an option, we could change it to `--exclude-devdependencies` or something?

helio-frota replied:
Also using the alias 'edd' 
`--exclude-dev-dependencies` or `--edd` just to follow the pattern of the others

helio-frota replied:
Looking great, and yes, if the default behavior is to use production only it will save the users to add 1 more option , making the command call more clean. +1

grdryn commented on test/lib/checker-test.js:
```diff
@@ -4,17 +4,36 @@ const test = require('blue-tape');
     t.ok(json[0].hasOwnProperty('dependency'), `The json has dependency property.`);
     t.ok(json[0].hasOwnProperty('licenses'), `The json has licenses property.`);
     t.ok(json[0].hasOwnProperty('file'), `The json has file property.`);
+    t.ok(numNycDeps(json) > 0, `Contains devDependency 'nyc'`);
```
Not sure how best to test this...the test depending on a particular devDependency seems a bit brittle.

helio-frota commented on lib/modules/checker.js:
```diff
@@ -2,9 +2,9 @@
+const check = (cwd, productionOnly) => {
   return new Promise((resolve, reject) => {
-    licenseChecker.init({start: cwd}, (error, json) => {
+    licenseChecker.init({start: cwd, production: productionOnly}, (error, json) => {
```
Awesome!

grdryn commented:
@helio-frota I've changed it now to only include runtime dependencies by default, and added an optional `--include-dev-dependencies`/`--idd` option that can be set to get the old behaviour.

helio-frota commented:
@grdryn great! thanks!
